include/ck_tile/ops/gemm/kernel/gemm_kernel.hpp

```diff
@@ -160,9 +160,11 @@ struct GemmKernel
     CK_TILE_HOST static bool IsSupportedArgument(const GemmKernelArgs& kargs)
     {
         constexpr bool is_output_c_reg_transposed =
-            EpiloguePipeline::IsOutputTransposed() != GemmPipeline::IsTransposeC();
-        if constexpr(!((GemmPipeline::VectorSizeC % 2 == 0 &&
-                        std::is_same_v<CLayout, tensor_layout::gemm::RowMajor> &&
+            EpiloguePipeline::IsOutputTransposed() || GemmPipeline::TransposeC();
+        if constexpr(!((GemmPipeline::template GetVectorSizeC<
+                                EpiloguePipeline::IsOutputTransposed()>() %
+                                2 ==
+                            0 &&
```

Comment:
You shouldn't here mix pipeline with epilogue. You can get information from both and then deduce what you need.

Reply:
So how can we deduce vector size without knowledge about layout after epliogue? We can discuss it